x-pack/plugins/observability_solution/profiling/public/plugin.ts

```diff
@@ -64,29 +65,47 @@ export class ProfilingPlugin
     ];
 
     const kuerySubject = new BehaviorSubject<string>('');
+    const appUpdater$ = new BehaviorSubject<AppUpdater>(() => ({}));
 
     const section$ = combineLatest([from(coreSetup.getStartServices()), kuerySubject]).pipe(
       map(([[coreStart], kuery]) => {
         if (coreStart.application.capabilities.profiling.show) {
-          const sections: NavigationSection[] = [
-            {
-              label: i18n.translate('xpack.profiling.navigation.sectionLabel', {
-                defaultMessage: 'Universal Profiling',
-              }),
-              entries: links.map((link) => {
-                return {
-                  app: OBLT_PROFILING_APP_ID,
-                  label: link.title,
-                  path: `${link.path}?kuery=${kuery ?? ''}`,
-                  matchPath: (path) => {
-                    return path.startsWith(link.path);
-                  },
-                };
-              }),
-              sortKey: 700,
-            },
-          ];
-          return sections;
+          let isSidebarEnabled = true;
```

Comment:
Nit: I am wondering if this will be better as a private variable similar to [here](https://github.com/elastic/kibana/blob/d9c1cd30af82abf12b5d569cbaa8886fbb78a5bb/x-pack/plugins/search_solution/search_navigation/public/plugin.ts#L51-L52) where it's also unsubscribed in the stop method and it can be reused in the future but that's not a must here. Of course up to you to pick the approach here, I saw some variations already in different plugins

Reply:
@jennypavlova  I've been considering this and other approaches. Ultimately, I decided to avoid keeping the observable data stream open for the entire lifetime of the plugin, since we need the information only in the particular situation (updating navigation).

My reasoning behind this was to avoid unnecessary code execution in subscribe callback. Yet, very simple because it's only variable assignment. On the other hand, I support your argument about easier reusability. 

I regret we don't have any helper or service methods to differentiate between navigation types. That would solve the problem and prevent multiple approaches to the same thing 😅 

Please raise a hand if you still insist on our proposal. I'll update the code then. Thanks for solid review 😀

Reply:
I agree, we can keep it and if we have a case where we need to reuse it we can update it.